pkg/directory/okta/provider.go

```diff
@@ -2,266 +2,172 @@ package okta
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
 	"fmt"
-	"io"
-	"net/http"
-	"net/url"
 	"sort"
-	"strconv"
+	"sync"
 	"time"
 
-	"github.com/tomnomnom/linkheader"
+	"github.com/okta/okta-sdk-golang/v2/okta"
+	"github.com/okta/okta-sdk-golang/v2/okta/query"
+	"golang.org/x/exp/maps"
 
 	"github.com/pomerium/datasource/pkg/directory"
 )
 
 // A Provider is an Okta user group directory provider.
 type Provider struct {
-	cfg         *config
-	lastUpdated *time.Time
-	groups      map[string]directory.Group
+	cfg *config
+
+	mu           sync.Mutex
+	groups       map[string]okta.Group
+	groupMembers map[string][]okta.User
 }
 
 // New creates a new Provider.
 func New(options ...Option) *Provider {
 	return &Provider{
-		cfg:    getConfig(options...),
-		groups: make(map[string]directory.Group),
+		cfg: getConfig(options...),
+
+		groups:       make(map[string]okta.Group),
+		groupMembers: make(map[string][]okta.User),
 	}
 }
 
-// GetDirectory fetches the groups of which the user is a member
-// https://developer.okta.com/docs/reference/api/users/#get-user-s-groups
+// GetDirectory gets the full directory information for Okta.
 func (p *Provider) GetDirectory(ctx context.Context) ([]directory.Group, []directory.User, error) {
-	groups, err := p.getGroups(ctx)
+	ctx, client, err := okta.NewClient(ctx,
+		okta.WithHttpClientPtr(p.cfg.getHTTPClient()),
+		okta.WithOrgUrl(p.cfg.url),
+		okta.WithTestingDisableHttpsCheck(true),
```

Comment:
accidentally left in?

Reply:
ok I switched this to be a config option so we can only add it for tests